src/Compilers/Core/MSBuildTask/ValidateBootstrap.cs

```diff
@@ -74,6 +89,21 @@ private static string NormalizePath(string path)
         }
 
         private string GetDirectory(Assembly assembly) => Path.GetDirectoryName(Utilities.GetLocation(assembly));
+
+        internal static void AddFailedLoad(AssemblyName name)
+        {
+            switch (name.Name)
+            {
+                case "System":
+                case "System.Core":
+                case "Microsoft.Build.Tasks.CodeAnalysis.resources":
+                    // These are failures are expected by design.
```

Comment:
Asked @tmat earlier and haven't heard back.  Assumption is that on desktop we don't redirect these because they are custom redirected my mscorlib.  IT has special knowledge about them